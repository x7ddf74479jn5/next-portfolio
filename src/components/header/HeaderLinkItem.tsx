import Link from "next/link";
import styles from "src/styles/components/header/HeaderLink.module.scss";

type Item = {
  label: string;
  href: string;
  icon?: {
    src: string;
    alt: string;
  };
};

type Prop = {
  item: Item;
};
const HeaderLinkItem: React.VFC<Prop> = ({ item }) => {
  return "icon" in item ? (
    <Link href={item.href}>
      <a className={styles.linkItem}>
        <img src={item.icon?.src} alt={item.icon?.alt} />
        <span>{item.label}</span>
      </a>
    </Link>
  ) : (
    <Link href={item.href}>
      <a className={styles.linkItem}>{item.label}</a>
    </Link>
  );
};

export default HeaderLinkItem;
