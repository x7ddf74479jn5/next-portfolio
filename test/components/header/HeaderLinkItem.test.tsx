import HeaderLinkItem from "src/components/header/HeaderLinkItem";
import type { RenderResult } from "test/test-utils";
import { reTestCase } from "test/test-utils";
import { render } from "test/test-utils";

describe("HeaderLinkItem", () => {
  let renderResult: RenderResult;

  describe("with icon", () => {
    const icon = {
      src: "img.png",
      alt: "alt",
    };
    const item = {
      label: "label",
      href: "href",
      icon: icon,
    };
    beforeEach(() => {
      renderResult = render(<HeaderLinkItem item={item} />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      const anchorElement = renderResult.getByRole("link");
      expect(anchorElement).toHaveAttribute("href", reTestCase.anyWord);
      expect(anchorElement).toHaveTextContent("label");

      const imgElement = renderResult.getByRole("img");
      expect(imgElement).toHaveAttribute("alt", reTestCase.anyWord);
      expect(imgElement).toHaveAttribute("src", reTestCase.anyImage);
    });
  });

  describe("no icon", () => {
    const item = {
      label: "label",
      href: "href",
    };
    beforeEach(() => {
      renderResult = render(<HeaderLinkItem item={item} />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      const imgElement = renderResult.queryByRole("img");
      expect(imgElement).toBeNull();
    });
  });
});
