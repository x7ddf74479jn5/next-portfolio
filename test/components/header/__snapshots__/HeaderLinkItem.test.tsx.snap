// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLinkItem no icon matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="linkItem"
    href="/href"
  >
    label
  </a>
</DocumentFragment>
`;

exports[`HeaderLinkItem with icon matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="linkItem"
    href="/href"
  >
    <img
      alt="alt"
      src="img.png"
    />
    <span>
      label
    </span>
  </a>
</DocumentFragment>
`;
